source/dotnet/Test/AdaptiveCards.Test/SerializationTests.cs

```diff
@@ -20,7 +20,7 @@ public void TestCardsSerializeInTheCorrectOrder()
             card.Version = "1.0";
             card.FallbackText = "Fallback Text";
             card.Speak = "Speak";
-            card.BackgroundImage = new Uri("http://adaptivecards.io/content/cats/1.png");
+            card.BackgroundImage = new AdaptiveBackgroundImage("http://adaptivecards.io/content/cats/1.png");
             card.Body.Add(new AdaptiveTextBlock { Text = "Hello" });
```

Comment:
Can we get some test coverage for the other properties with the background image serialization as well as for the containers?

Reply:
Testing them with the JSON tests in #2316.

Reply:
Added test on line 425.

---
In reply to: [250430169](https://github.com/Microsoft/AdaptiveCards/pull/2234#discussion_r250430169) [](ancestors = 250430169)

Reply:
This is a breaking change to the API, we should be mindful of these. I pushed a fix that adds implicit conversion from Uri to the new AdaptiveBackgroundImage image type